src/test/java/hyperledger/besu/java/rest/client/TransactionServiceTest.java

```diff
@@ -1,28 +1,146 @@
+import org.junit.jupiter.api.TestInstance;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.mock.web.MockMultipartFile;
 import org.testcontainers.shaded.org.apache.commons.io.FileUtils;
 import org.web3j.EVMTest;
 import org.web3j.NodeType;
+import org.web3j.abi.datatypes.Address;
+import org.web3j.abi.datatypes.generated.Uint256;
+
+import java.io.File;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+
+import static hyperledger.besu.java.rest.client.exception.ErrorCode.HYPERLEDGER_BESU_NO_ABI_DEFINITION_FOUND_ERROR;
 
 @EVMTest(type = NodeType.BESU)
 @SpringBootTest
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
 public class TransactionServiceTest {
 
-  private static final String compiledContract = "compiled-smart-contract";
+    private static final String getputContractByteCode = "src/test/resources/get-put-state-bytecode";
+    private static final String NFTByteCode = "src/test/resources/NFT-bytecode";
+
+    @Autowired
+    private TransactionService transactionService;
+    private MockMultipartFile mockMultipartFile;
+    private String functionName;
+    private String contractAddress;
+    private String contractAddressGetPut;
+    private String contractAddressNFT;
+
+    @BeforeAll
+    void init() throws Exception {
+        String compiledCode = FileUtils.readFileToString(new File(getputContractByteCode), StandardCharsets.UTF_8);
+        ClientResponseModel response = transactionService.deploy(compiledCode);
+        contractAddressGetPut = (String) response.getMessage();
+        Assertions.assertEquals(response.getCode(), ErrorConstants.NO_ERROR);
+
+        String NFTCode = FileUtils.readFileToString(new File(NFTByteCode), StandardCharsets.UTF_8);
+        response = transactionService.deploy(NFTCode);
+        contractAddressNFT = (String) response.getMessage();
+
+    }
 
-  @Autowired private TransactionService transactionService;
+    @Test
+    void testDeployment() throws IOException {
+        String compiledCode = FileUtils.readFileToString(new File(getputContractByteCode), StandardCharsets.UTF_8);
+        ClientResponseModel response = transactionService.deploy(compiledCode);
+        Assertions.assertEquals(response.getCode(), ErrorConstants.NO_ERROR);
+        Assertions.assertNotNull(response.getMessage());
+    }
 
-  @Test
-  void testDeploymet() throws IOException {
-    String compiledCode =
-        FileUtils.readFileToString(new File(compiledContract), StandardCharsets.UTF_8);
-    transactionService.deploy(compiledCode);
-  }
+    @Test
+    void testExecutionBurn() throws Exception {
+        mockMultipartFile = new MockMultipartFile("NFT.abi", Files.newInputStream(new File("src/test/resources/NFT.abi").toPath()));
+        functionName = "burn";
+        Object[] params = {new Uint256(BigInteger.valueOf(1))};
```

Comment:
new contract will accept string as the parameter

Reply:
Yes, these changes are part of https://github.com/arsulegai/eth-connector/pull/10, wasn't aware about the order in which PR's will be merged, so didn't add it here. Will amend once the other PR gets merged or if this gets merged earlier, will create a new PR to accomodate with the new requirements.

Reply:
Updated all the tests to accept string.